wafer/talks/views.py

```diff
@@ -141,3 +146,35 @@ def get_context_data(self, **kwargs):
             user__talks__status='A').distinct().prefetch_related('user')
         context["speaker_rows"] = self._by_row(speakers, 4)
         return context
+
+
+class TalksViewSet(viewsets.ModelViewSet):
+    """API endpoint that allows talks to be viewed or edited."""
+    queryset = Talk.objects.all()
+    serializer_class = TalkSerializer
+    permission_classes = (DjangoModelPermissions, )
+
+
+    def list(self, request):
+        # We override the default implementation to only show accepted talks
+        # to people who aren't part of the management group
+        if Talk.can_view_all(request.user):
+            queryset = Talk.objects.all()
```

Comment:
It's probably safer to do this by defining `def get_queryset(self, request):`. To use the permissions class, I think then one has to keep `queryset = Talk.objects.none()` so that the permissions understand which model class is being used.